src/features/auth/utils/registerFormSchema.ts

```diff
@@ -0,0 +1,58 @@
+import { t } from 'i18next';
+import { z } from 'zod';
+import { toFormikValidationSchema } from 'zod-formik-adapter';
+
+import { translationWithPrefix } from '@/utils/translationWithPrefix';
+
+import {
+  FULL_NAME_MAX_LENGTH,
+  FULL_NAME_MIN_LENGTH,
+  PASSWORD_REGEX,
+  USERNAME_MAX_LENGTH,
+  USERNAME_MIN_LENGTH,
+} from './constants';
+
+/* schema */
+export const registerFormSchema = z.object({
+  userId: z.string().email(),
+  fullName: z
+    .string()
+    .min(FULL_NAME_MIN_LENGTH)
+    .max(FULL_NAME_MAX_LENGTH),
+  userName: z
+    .string()
+    .min(USERNAME_MIN_LENGTH)
+    .max(USERNAME_MAX_LENGTH),
+  password: z.string().regex(PASSWORD_REGEX),
+});
+
+/* schema adapted to formik */
+export const registerFormikSchema = toFormikValidationSchema(registerFormSchema);
+
+/* localized error messages */
+const tp = translationWithPrefix('error.validations');
+
+const customErrorMap: z.ZodErrorMap = (issue, ctx) => {
```

Comment:
If feel like you can put this logic in the registerFormSchema using refinement and custom errors:
`z.string().min(5, { message: "Must be 5 or more characters long" });`

`const myString = z.string().refine((val) => val.length <= 255, {
  message: "String can't be more than 255 characters",
});`

Reply:
That's a great suggestion. I attempted it like this at first but was getting errors from z and typescript. 

I finally figured it out :)